nevergrad/functions/pyomo/core.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import typing as tp
+from functools import partial
+import types
+import numpy as np
+from nevergrad.parametrization import parameter as p
+import pyomo.environ as pyo
+from pyomo.core.expr import current as pyo_expr
+#from . import photonics
+from .. import base
+
+def _make_pyomo_range_set_to_parametrization(pyo_model : pyo.Model, model_component : pyo.RangeSet, params : dict):
+    # https://pyomo.readthedocs.io/en/stable/pyomo_modeling_components/Sets.html
+    # Refer to the implementation in pyomo/core/base/set.py
+    if not((model_component.dimen == 1) and (len(model_component.ranges()) == 1)):
+        raise NotImplementedError("RangeSet setting is currently not implemented")
+    for i, r in enumerate(model_component.ranges()):
+        params[model_component.name + '[' + str(i) + ']'] = p.Scalar(lower=r[0], upper=r[1])
+    return params
+
+def _make_pyomo_variable_to_parametrization(pyo_model : pyo.Model, model_component : pyo.Var, params : dict):
+    # https://pyomo.readthedocs.io/en/stable/pyomo_modeling_components/Sets.html
+    # Refer to the implementation in pyomo/core/base/var.py
+    if not model_component._data:
+        return
+    for k, v in model_component._data.items():
+        params[model_component.name + '[' + str(k) + ']'] = p.Scalar(lower=v.bounds[0], upper=v.bounds[1])
+    return params
+
+class Pyomo(base.ExperimentFunction):
+    """Function calling photonics code
+
+    Parameters
+    ----------
+    name: str
+        problem name
+    model: pyomo.environ.model
+        Pyomo model
+
+    Returns
+    -------
+    float
+        the fitness
+
+    Notes
+    -----
+    - You will require an Pyomo installation (with pip: "pip install pyomo")
+    - Any changes on the model externally can lead to unexpected behaviours.
+    """
+
+    def __init__(self, name : str, model : pyo.Model) -> None:
```

Comment:
if the name is not needed you can remove it.

Reply:
The name is kept.